src/ahp/indexer.rs

```diff
@@ -149,13 +151,13 @@ impl<F: PrimeField> AHPForR1CS<F> {
             cs: PhantomData,
         };
 
-        let domain_h = GeneralEvaluationDomain::new(num_constraints)
+        let domain_h = EvaluationDomain::new(num_constraints)
             .ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
-        let domain_k = GeneralEvaluationDomain::new(num_non_zero)
+        let domain_k =
+            EvaluationDomain::new(num_non_zero).ok_or(SynthesisError::PolynomialDegreeTooLarge)?;
```

Comment:
On a general note, should we make `EvaluationDomain::new` return a Result instead of an option instead? Probably in a separate PR in the future